platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
```

Comment:
Allowing timeout would be an extra feature. POSIX doesn't provide that - the mechanism is to use the timeout on poll() if necessary. I'd like to keep complexity down, given that we're expecting FileHandle implementers to do it, unlike NSAPI where it's the common code.